aten/src/ATen/native/quantized/cpu/qnnpack/src/fc-prepack.cc

```diff
@@ -0,0 +1,53 @@
+#include <pytorch_qnnpack.h>
+#include <qnnpack/pack.h>
+#include <qnnpack_func.h>
+#include <cstring>
+
+namespace qnnpack {
+PackBMatrix::PackBMatrix(
+    size_t input_channels,
+    size_t output_channels,
+    uint8_t kernel_zero_point,
+    float kernel_scale,
+    const uint8_t* kernel,
+    const int32_t* bias) {
+  if (kernel_scale <= 0.0f || !std::isnormal(kernel_scale)) {
+    pytorch_qnnp_log_error(
+        "failed to create fully connected operator with %.7g kernel scale: "
+        "scale must be finite and positive",
+        kernel_scale);
+    assert("QNNPACK Runtime Error.");
+  }
+
+  const uint32_t nr = pytorch_qnnp_params.q8conv.nr;
+  const uint32_t kr = pytorch_qnnp_params.q8conv.kr;
+
+  const uint32_t n_stride = (output_channels + (nr - 1)) & -nr;
+  const uint32_t k_stride = (input_channels + (kr - 1)) & -kr;
```

Comment:
For this computation to be correct, nr and kr must be powers of two themselves so I think it is worthwhile to check that assumption at some level - be it in the caller or here.

Reply:
These parameters are set in the init functions of QNNPACK, I don't think the user edits them. But I can still add the check.

Reply:
You are right!